### PR TITLE
Release v0.7.0 and Bump Versions

### LINKED ISSUE #834: Feast 0.7 Release
Kicking off this issue to document the upcoming 0.7 release.

At the moment the roadmap is not 100% solidified yet, but we are proposing the following:

#### New Functionality
1. Entities as a first-class concept #405 
1. ~Datasets as a first-class concept~ (pushed out)
1. Feast UI (MVP)
1. Native SDK types instead of proto types
1. Audit log support
1. Request/response logging for online serving

#### Technical debt, refactoring, or housekeeping
1. Improved integration testing framework
1. Rectify all flaky batch tests
1. Decouple job management from Feast Core
1. More descriptive error logs in Feast online serving

#### Proposals
1. Training-serving skew detection proposal

### PR DESCRIPTION
**Pull request by mrzzy**
**What this PR does / why we need it**:
New Release v0.7.0:
- Updates `CHANGELOG.md`  v0.7.0 changelog.
- Update versions in docs to point to v0.7.0.
- Bump versions on master to v0.8-SNAPSHOT.

**Which issue(s) this PR fixes**:

Fixes #834 

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: infra/scripts/validate-version-consistency.sh [hunk 1]
```diff
@@ -123,7 +125,7 @@ for i in "${files_to_validate_version[@]}"; do
   echo "Testing whether versions are correctly set within file: $FILE_PATH"
   ACTUAL_OCCURRENCES=$(grep -c -P "\bv?$VERSION\b" "$FILE_PATH" || true)
 
-  if [ "${ACTUAL_OCCURRENCES}" -eq "${EXPECTED_OCCURRENCES}" ]; then
+  if [ "${ACTUAL_OCCURRENCES}" -ge "${EXPECTED_OCCURRENCES}" ]; then
```
Comment (mrzzy) [inline infra/scripts/validate-version-consistency.sh:128]: Actual occurrences may be greater that expected occurrences when both pom.xml and tag versions the same. This is required as lint flags any occurrence of the version.

**Review by woop — APPROVED** [review]

**Pull request merged**